### DISCUSSION TITLE
Create one line agent install script

### DISCUSSION THREAD

**Discussion by asheroto** [amidaware/tacticalrmm#885] [Ideas]
I have a handful of computers in different locations that need agents installed. I am using ScreenConnect already. At first I was going to each machine, transferring the agent EXE, then running it, but found this tedious for many computers. I wrote a quick PowerShell script that will take the agent EXE's URL and generate a PowerShell script to download and install it. It converts the script into an encoded command (base64) so that it can be easily copy-and-pasted. It also adds a temporary AV exception so that the agent installer isn't flagged (I found this to be a problem without doing so).

A great feature would be adding something like this to TacticalRMM natively so you could just copy-and-paste a single line command to install the agent for the location.

Gist of my PowerShell script is here:
https://gist.github.com/asheroto/162e396e8b041bea2f90fe3d8ff94348

**Comment by silversword411** [comment]
There is a simple native script to install TRMM using the screenconnect command interface as well:
https://wh1te909.github.io/tacticalrmm/3rdparty_screenconnect/#install-tactical-rmm-via-screeconnect-commands-window

**Reply by Darkpriest258** [reply]
do you still use screenconnect?

**Comment by asheroto** [comment]
Gotcha. The main reason I didn't use that is because it doesn't exclude AV for Windows Defender when downloading/installing the agent. The agent installer adds Windows Defender exclusions, but not for the agent installer itself.

**Reply by silversword411** [reply]
I'm code signed, so never have a problem...unsigned you might need some exclusions in there as well:
```powershell
powershell Add-MpPreference -ExclusionPath C:\TEMP\TRMM
    powershell Add-MpPreference -ExclusionPath "C:\Program Files\TacticalAgent\*"
    powershell Add-MpPreference -ExclusionPath C:\Windows\Temp\winagent-v*.exe
    powershell Add-MpPreference -ExclusionPath "C:\Program Files\Mesh Agent\*"
    powershell Add-MpPreference -ExclusionPath C:\Windows\Temp\TRMM\*
```

**Comment by asheroto** [comment]
Gotcha.

Those exclusions are fortunately added after the agent is installed by the agent installer.
